scripts/convert_olmo_to_hf_new.py

```diff
@@ -203,7 +203,14 @@ def _write_tokenizer(
     if input_tokenizer_path is not None:
         base_tokenizer = Tokenizer.from_file(str(input_tokenizer_path))
     else:
-        base_tokenizer = Tokenizer.from_pretrained(checkpoint_dir)
+        config_path = Path(checkpoint_dir) / "config.yaml"
+        tokenizer_config = yaml.safe_load(config_path.read_text())["tokenizer"]
+
+        # Initialize tokenizer and validate vocab size.
+        if Path(tokenizer_config["identifier"]).is_file():
+            base_tokenizer = Tokenizer.from_file(tokenizer_config["identifier"])
+        else:
+            base_tokenizer = Tokenizer.from_pretrained(tokenizer_config["identifier"])
```

Comment:
Any reason not to do this:

```suggestion
        base_tokenizer = Tokenizer.from_checkpoint(checkpoint_dir)
```

Reply:
This is `Tokenizer` from the `tokenizers` library, not the one from the OLMo code base. I made my first PR not remembering this.

Reply:
Ah, ok